internal/pkg/log/log.go

```diff
@@ -0,0 +1,83 @@
+package log
+
+import (
+	"octavius/internal/pkg/constant"
+	"os"
+	"time"
+
+	"github.com/rs/zerolog"
+)
+
+type engine struct {
+	logger *zerolog.Logger
+	// put configurabe field here
+}
+
+var logEngine engine // contain cli configarution
+
+// Init intializes the logger object
+// TODO: cater logger requirement for cli and control plane
+func Init(configLogLevel string, logFile string) error {
+
+	var (
+		f     *os.File
+		err   error
+		multi zerolog.LevelWriter
+	)
+
+	logLevel, err := zerolog.ParseLevel(configLogLevel)
+	if err != nil {
+		return err
+	}
+
+	consoleWriter := zerolog.ConsoleWriter{
+		Out: os.Stdout,
+	}
+
+	if logFile != "" {
+		f, err = os.OpenFile(logFile, os.O_RDWR|os.O_CREATE|os.O_APPEND, 0666)
+		if err != nil {
+			return err
+		}
+		multi = zerolog.MultiLevelWriter(f, consoleWriter)
+	}
+
+	zerolog.TimeFieldFormat = time.RFC850
+
+	zerologInstance := zerolog.New(multi).With().Timestamp().CallerWithSkipFrameCount(constant.LoggerSkipFrameCount).Logger().Level(logLevel)
+	logEngine = engine{
+		logger: &zerologInstance,
+	}
+
+	return nil
+}
+
+//Debug logs the message at debug level
+func Debug(msg string) {
+	logEngine.logger.Debug().Msg(msg)
+}
+
+//Info logs the message at info level
+func Info(msg string) {
+	logEngine.logger.Info().Msg(msg)
+}
+
+//Warn logs the message at warn level
+func Warn(msg string) {
+	logEngine.logger.Warn().Msg(msg)
+}
+
+//Error logs the message and error at error level if err is not nil else it logs message at info level
+func Error(err error, msg string) {
+	logEngine.logger.Err(err).Msg(msg)
+}
+
+//Fatal logs the message at fatal level followed by a os.Exit(1) call
+func Fatal(msg string) {
+	logEngine.logger.Fatal().Msgf(msg)
+}
+
+//Panic logs the message and error at panic level and stops the ordinary flow of goroutine
+func Panic(msg string, err error) {
+	logEngine.logger.Panic().Err(err).Msg(msg)
+}
```

Comment:
The cli logger earlier was also doing color printing of messages of the logs, in a user centric format, given now we have a common logger, we have two options for color printing at cli end, print the messages wherever the event occurs itself, or have if-else structure inside logs that if the mode is cli, print the message, I guess first option would be better thing. What are your views?

Reply:
Let's focus on logger in another PR, this will only cover the code structuring and some refactoring. I think we can add option to use color printing in the log if it's needed, that's why I put a TODO comment on top of this file to cater to the requirement for CLI.

Reply:
Okay! Sure, sounds fine.